### ISSUE TITLE
No validation of nil module_path

### ISSUE THREAD

**Issue by jeff1evesque** [jantman/vagrant-r10k#18]
When trying to `vagrant up` on a windows 7 host machine, and centos 6.6 guest VM, using vagrant-r10k, we get the following output:

```bash
$ vagrant up
Bringing machine 'default' up with 'virtualbox' provider...
==> default: vagrant-r10k: Building the r10k module path with puppet provisioner module_path "". (if module_path is an array, first element is used)
==> default: vagrant-r10k: Building the r10k module path with puppet provisioner module_path "". (if module_path is an array, first element is used)
C:/Users/Jeff/.vagrant.d/gems/gems/vagrant-r10k-0.2.0/lib/vagrant-r10k/modulegetter.rb:90:in `join': no implicit conversion of nil into String (TypeError)
        from C:/Users/Jeff/.vagrant.d/gems/gems/vagrant-r10k-0.2.0/lib/vagrant-r10k/modulegetter.rb:90:in `call'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:95:in `block in finalize_action'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action warden.rb:34:in `call'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/builtin/handle_box.rb:56:in `call'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:95:in `block in finalize_action'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/builder.rb:116:in `call'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/runner.rb:66:in `block in run'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/util/busy.rb:19:in `busy'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/runner.rb:66:in `run'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/builtin/call.rb:53:in `call'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/plugins/providers/virtualbox/action/check_virtualbox.rb:17:in `call'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
        from C:/Users/Jeff/.vagrant.d/gems/gems/vagrant-triggers-0.5.1/lib/vagrant-triggers/action/trigger.rb:17:in `call'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
        from C:/Users/Jeff/.vagrant.d/gems/gems/vagrant-triggers-0.5.1/lib/vagrant-triggers/action/trigger.rb:17:in `call'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
        from C:/Users/Jeff/.vagrant.d/gems/gems/vagrant-triggers-0.5.1/lib/vagrant-triggers/action/trigger.rb:17:in `call'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/builder.rb:116:in `call'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/runner.rb:66:in `block in run'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/util/busy.rb:19:in `busy'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/runner.rb:66:in `run'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/machine.rb:214:in `action_raw'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/machine.rb:191:in `block in action'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/environment.rb:516:in `lock'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/machine.rb:178:in `call'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/machine.rb:178:in `action'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant batch_action.rb:82:in `block (2 levels) in run'
```

The following are the pertinent files from our [drupal-demonstration](https://github.com/jeff1evesque/drupal-demonstration) repository:

- `Vagrantfile`: run without being versioned into github -

```
# -*- mode: ruby -*-
# vi: set ft=ruby :

# All Vagrant configuration is done below. The "2" in Vagrant.configure
# configures the configuration version (we support older styles for
# backwards compatibility). Please don't change it unless you know what
# you're doing.
Vagrant.configure(2) do |config|
  # The most common configuration options are documented and commented below.
  # For a complete reference, please see the online documentation at
  # https://docs.vagrantup.com.

  ## Variables (ruby syntax)
  required_plugins = %w(vagrant-r10k vagrant-vbguest)
  plugin_installed = false

  ## Install Vagrant Plugins
  required_plugins.each do |plugin|
    unless Vagrant.has_plugin? plugin
      system "vagrant plugin install #{plugin}"
      plugin_installed = true
    end
  end

  ## Restart Vagrant: if new plugin installed
  if plugin_installed == true
    exec "vagrant #{ARGV.join(' ')}"
  end

  # Every Vagrant development environment requires a box. You can search for
  # boxes at https://atlas.hashicorp.com/search.
  config.vm.box = "puppetlabs/centos-6.6-64-puppet"
  config.vm.box_version = "1.0.1"

  ## Run r10k
  config.r10k.puppet_dir = 'puppet'
  config.r10k.puppetfile_path = 'puppet/Puppetfile'

  # Define fully qualified domain name
  config.vm.hostname = "drupal-demonstration.com"

  # Custom Manifest: general configuration
  #
  #  Note: future parser allow array iteration in the puppet manifest
  config.vm.provision "puppet" do |puppet|
    puppet.manifests_path = "puppet/manifests"
    puppet.manifest_file  = "default.pp"
    puppet.options        = ["--parser", "future"]
  end

  # Custom Manifest: ssl configuration
  config.vm.provision "puppet" do |puppet|
    puppet.manifests_path = "puppet/manifests"
    puppet.manifest_file  = "ssl.pp"
  end

  # Disable automatic box update checking. If you disable this, then
  # boxes will only be checked for updates when the user runs
  # `vagrant box outdated`. This is not recommended.
  # config.vm.box_check_update = false

  # Create a forwarded port mapping which allows access to a specific port
  # within the machine from a port on the host machine. In the example below,
  # accessing "localhost:8080" will access port 80 on the guest machine.
  config.vm.network "forwarded_port", guest: 80, host: 6060
  config.vm.network "forwarded_port", guest: 443, host: 6161

  # Create a private network, which allows host-only access to the machine
  # using a specific IP.
  # config.vm.network "private_network", ip: "192.168.33.10"

  # Create a public network, which generally matched to bridged network.
  # Bridged networks make the machine appear as another physical device on
  # your network.
  # config.vm.network "public_network"

  # Share an additional folder to the guest VM. The first argument is
  # the path on the host to the actual folder. The second argument is
  # the path on the guest to mount the folder. And the optional third
  # argument is a set of non-required options.
  # config.vm.synced_folder "../data", "/vagrant_data"

  # Provider-specific configuration so you can fine-tune various
  # backing providers for Vagrant. These expose provider-specific options.
  # Example for VirtualBox:
  #
  # config.vm.provider "virtualbox" do |vb|
  #   # Display the VirtualBox GUI when booting the machine
  #   vb.gui = true
  #
  #   # Customize the amount of memory on the VM:
  #   vb.memory = "1024"
  # end
  #
  # View the documentation for the provider you are using for more
  # information on available options.

  # Define a Vagrant Push strategy for pushing to Atlas. Other push strategies
  # such as FTP and Heroku are also available. See the documentation at
  # https://docs.vagrantup.com/v2/push/atlas.html for more information.
  # config.push.define "atlas" do |push|
  #   push.app = "YOUR_ATLAS_USERNAME/YOUR_APPLICATION_NAME"
  # end

  # Enable provisioning with a shell script. Additional provisioners such as
  # Puppet, Chef, Ansible, Salt, and Docker are also available. Please see the
  # documentation for more information about their specific syntax and use.
  # config.vm.provision "shell", inline: <<-SHELL
  #   sudo apt-get update
  #   sudo apt-get install -y apache2
  # SHELL
end
```

- [`Puppetfile`](https://github.com/jeff1evesque/drupal-demonstration/blob/af4c1ef00b8fd3bf9cba6636ff76e50f64236cff/puppet/Puppetfile)
- [`default.pp`](https://github.com/jeff1evesque/drupal-demonstration/blob/8e41afde6fd164c608cdc1d00b600084a0c4c30e/puppet/manifests/default.pp)
- [`ssl.pp`](https://github.com/jeff1evesque/drupal-demonstration/blob/af4c1ef00b8fd3bf9cba6636ff76e50f64236cff/puppet/manifests/ssl.pp)

**Comment by jantman** [comment]
@jeff1evesque can you please include the relevant versions and debug output asked for in https://github.com/jantman/vagrant-r10k#getting-help ?

Also, do you have rvm or rbenv configured and in use? I believe I've seen issues like this personally if I accidentally had rvm set to use a non-default ruby (i.e. other than the one Vagrant expects).

**Comment by jeff1evesque** [comment]
```
$ vagrant plugin list
vagrant-r10k (0.2.0)
vagrant-share (1.1.4, system)
vagrant-vbguest (0.10.0)
$ vagrant --version
Vagrant 1.7.4
$ vagrant ssh
$ ruby --version
sh.exe": ruby: command not found
```


```bash
$ VAGRANT_LOG=debug vagrant up
 INFO runner: Preparing hooks for middleware sequence...
 INFO runner: 4 hooks defined.
 INFO runner: Running action: authenticate_box_url #<Vagrant::Action::Builder:0x2eb9450>
 INFO warden: Calling IN action: #<VagrantPlugins::LoginCommand::AddAuthentication:0x456e070>
DEBUG client: No authentication token in environment or C:/Users/Jeff/.vagrant.d
/data/vagrant_login_token
 INFO warden: Calling OUT action: #<VagrantPlugins::LoginCommand::AddAuthenticat
ion:0x456e070>
 INFO loader: Set :"30600708_puppetlabs/centos-6.6-64-puppet_virtualbox" = #<Pathname:C:/Users/Jeff/.vagrant.d/boxes/puppetlabs-VAGRANTSLASH-centos-6.6-64-puppet/1.0.1/virtualbox/Vagrantfile>
DEBUG loader: Populating proc cache for #<Pathname:C:/Users/Jeff/.vagrant.d/boxe
s/puppetlabs-VAGRANTSLASH-centos-6.6-64-puppet/1.0.1/virtualbox/Vagrantfile>
DEBUG loader: Load procs for pathname: C:/Users/Jeff/.vagrant.d/boxes/puppetlabs
-VAGRANTSLASH-centos-6.6-64-puppet/1.0.1/virtualbox/Vagrantfile
 INFO loader: Loading configuration in order: [:"30600708_puppetlabs/centos-6.6-
64-puppet_virtualbox", :home, :root, "24447696_machine_default"]
DEBUG loader: Loading from: 30600708_puppetlabs/centos-6.6-64-puppet_virtualbox
(evaluating)
DEBUG loader: Loading from: root (cache)
DEBUG loader: Configuration loaded successfully, finalizing and returning
DEBUG push: finalizing
 INFO machine: Initializing machine: default
 INFO machine:   - Provider: VagrantPlugins::ProviderVirtualBox::Provider
 INFO machine:   - Box: #<Vagrant::Box:0x44e6f98>
 INFO machine:   - Data dir: c:/Users/Jeff/web-projects/drupal-demonstration/.va
grant/machines/default/virtualbox
DEBUG virtualbox: Instantiating the driver for machine ID: nil
DEBUG base: Windows. Trying VBOX_INSTALL_PATH for VBoxManage
DEBUG base: VBOX_INSTALL_PATH value: C:\Program Files\Oracle\VirtualBox\
 INFO base: VBoxManage path: C:\Program Files\Oracle\VirtualBox\VBoxManage.exe
 INFO subprocess: Starting process: ["C:\\Program Files\\Oracle\\VirtualBox\\VBoxManage.exe", "--version"]
 INFO subprocess: Command not in installer, restoring original environment...
DEBUG subprocess: Selecting on IO
DEBUG subprocess: stdout: 4.3.28r100309
DEBUG subprocess: Waiting for process to exit. Remaining to timeout: 32000
DEBUG subprocess: Exit status: 0
DEBUG meta: Finding driver for VirtualBox version: 4.3.28
 INFO meta: Using VirtualBox driver: VagrantPlugins::ProviderVirtualBox::Driver::Version_4_3
DEBUG base: Windows. Trying VBOX_INSTALL_PATH for VBoxManage
DEBUG base: VBOX_INSTALL_PATH value: C:\Program Files\Oracle\VirtualBox\
 INFO base: VBoxManage path: C:\Program Files\Oracle\VirtualBox\VBoxManage.exe
 INFO machine: New machine ID: nil
DEBUG virtualbox: Instantiating the driver for machine ID: nil
DEBUG base: Windows. Trying VBOX_INSTALL_PATH for VBoxManage
DEBUG base: VBOX_INSTALL_PATH value: C:\Program Files\Oracle\VirtualBox\
 INFO base: VBoxManage path: C:\Program Files\Oracle\VirtualBox\VBoxManage.exe
 INFO subprocess: Starting process: ["C:\\Program Files\\Oracle\\VirtualBox\\VBoxManage.exe", "--version"]
 INFO subprocess: Command not in installer, restoring original environment...
DEBUG subprocess: Selecting on IO
DEBUG subprocess: stdout: 4.3.28r100309
DEBUG subprocess: Waiting for process to exit. Remaining to timeout: 32000
DEBUG subprocess: Exit status: 0
DEBUG meta: Finding driver for VirtualBox version: 4.3.28
 INFO meta: Using VirtualBox driver: VagrantPlugins::ProviderVirtualBox::Driver::Version_4_3
DEBUG base: Windows. Trying VBOX_INSTALL_PATH for VBoxManage
DEBUG base: VBOX_INSTALL_PATH value: C:\Program Files\Oracle\VirtualBox\
 INFO base: VBoxManage path: C:\Program Files\Oracle\VirtualBox\VBoxManage.exe
 INFO command: With machine: default (#<VagrantPlugins::ProviderVirtualBox::Provider:0x426c078 @logger=#<Log4r::Logger:0x426c048 @fullname="vagrant::provider::virtualbox", @outputters=[], @additive=true, @name="virtualbox", @path="vagrant::provider", @parent=#<Log4r::Logger:0x647778 @fullname="vagrant", @outputters=[#<Log4r::StderrOutputter:0x2a862c8 @mon_owner=nil, @mon_count=0, @mon_mutex=#Mutex:0x2a86238>, @name="stderr", @level=0, @formatter=#<Log4r::DefaultFormatter:0x2a8d360 @depth=7>, @out=#<IO:<STDERR>>>], @additive=true, @name="vagrant", @path="", @parent=#<Log4r::RootLogger:0x647388 @level=0, @outputters=[]>, @level=1, @trace=false>, @level=1, @trace=false>, @machine=#<Vagrant::Machine: default (VagrantPlugins::ProviderVirtualBox::Provider)>, @driver=#<VagrantPlugins::ProviderVirtualBox::Driver::Meta:0x2e82148 @logger=#<Log4r::Logger:0x3580fb0 fullname="vagrant::provider::virtualbox::meta", @outputters=[], @additive=true, @name="meta", @path="vagrant::provider::virtualbox", @parent=#<Log4r::Logger:0x426c048 @fullname="vagrant::provider::virtualbox", @outputters=[], @additive=true, @name="virtualbox", @path="vagrant::provider", @parent=#<Log4r::Logger:0x647778 @fullname=
"vagrant", @outputters=[#<Log4r::StderrOutputter:0x2a862c8 @mon_owner=nil, @mon_count=0, @mon_mutex=#<Mutex:0x2a86238>, @name="stderr", @level=0, @formatter=#<Log4r::DefaultFormatter:0x2a8d360 @depth=7>, @out=#<IO:<STDERR>>>], @additive=true, @name="vagrant", @path="", @parent=#<Log4r::RootLogger:0x647388 @level=0, @outputters=[]>, @level=1, @trace=false>, @level=1, @trace=false>, @level=1, @trace=false>, @interrupted=false, @vboxmanage_path="C:\\Program Files\\Oracle\\VirtualBox\\VBoxManage.exe", @uuid=nil, @version="4.3.28", @driver=#VagrantPlugins::ProviderVirtualBox::Driver::Version_4_3:0x3a6ee70 @logger=#<Log4r::Logger:0x427e318 @fullname="vagrant::provider::virtualbox_4_3", @outputters=[], @additive=true
, @name="virtualbox_4_3", @path="vagrant::provider", @parent=#<Log4r::Logger:0x647778 @fullname="vagrant", @outputters=[#<Log4r::StderrOutputter:0x2a862c8 @mon_owner=nil, @mon_count=0, @mon_mutex=#<Mutex:0x2a86238>, @name="stderr", @level=0
, @formatter=#<Log4r::DefaultFormatter:0x2a8d360 @depth=7>, @out=#<IO:<STDERR>>>], @additive=true, @name="vagrant", @path="", @parent=#<Log4r::RootLogger:0x647388 @level=0, @outputters=[]>, @level=1, @trace=false>, @level=1, @trace=false>,
@interrupted=false, @vboxmanage_path="C:\\Program Files\\Oracle\\VirtualBox\\VBoxManage.exe", @uuid=nil>>, @cap_logger=#<Log4r::Logger:0x442cd88 @fullname="vagrant::capability_host::vagrantplugins::providervirtualbox::provider", @outputters=[], @additive=true, @name="provider", @path="vagrant::capability_host::vagrantplugins::providervirtualbox", @parent=#<Log4r::Logger:0x647778 @fullname="vagrant
", @outputters=[#<Log4r::StderrOutputter:0x2a862c8 @mon_owner=nil, @mon_count=0,  @mon_mutex=#<Mutex:0x2a86238>, @name="stderr", @level=0, formatter=#Log4r::DefaultFormatter:0x2a8d360 @depth=7>, @out=#<IO:<STDERR>>>], @additive=true, @name="vagrant", @path="", @parent=#<Log4r::RootLogger:0x647388 @level=0, @outputters=[]>, @level=1, @trace=false>, @level=1, @trace=false>, @cap_host_chain=[[:virtualbox, #<#<Class:0x442d190>:0x445fae8>]], @cap_args=[#<Vagrant::Machine: default (VagrantPlugins::ProviderVirtualBox::Provider)>], @cap_caps={:docker=>#Vagrant::Registry:0x442d040 @items={:public_address=>#<Proc:0x369f210@c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/plugins/providers/docker/plugin.rb:54>, :proxy_machine=>#<Proc:0x369f0f0@c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/plugins/providers/docker/plugin.rb:59>}, @results_cache={}>, :hyperv=>#<Vagrant::Registry:0x442cf80 @items={:public_address=>#<Proc:0x36069f0@c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/plugins/providers/hyperv/plugin.rb:25>},
 @results_cache={}>, :virtualbox=>#<Vagrant::Registry:0x442cf08 @items={:forwarded_ports=>#<Proc:0x36d1ac0@c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/plugins/providers/virtualbox/plugin.rb:27>, :nic_mac_addresses=>#<Proc:0x36d1a60@c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/plugins/providers/virtualbox/plugin.rb:32>, :public_address=>#<Proc:0x3819af8@c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-share-1.1.4/lib/vagrant-share.rb:39>}, @results_cache={}>}
>)
 INFO interface: info: Bringing machine 'default' up with 'virtualbox' provider.
..
Bringing machine 'default' up with 'virtualbox' provider...
 INFO batch_action: Enabling parallelization by default.
 INFO batch_action: Disabling parallelization because provider doesn't support it: virtualbox
 INFO batch_action: Batch action will parallelize: false
 INFO batch_action: Starting action: #<Vagrant::Machine:0x381a5d8> up {:destroy_on_error=>true, :parallel=>true, :provision_ignore_sentinel=>false, :provision_types=>nil}
 INFO machine: Calling action: up on provider VirtualBox (new VM)
DEBUG environment: Attempting to acquire process-lock: machine-action-5e8fabf089da6737431eeafdb74f5224
DEBUG environment: Attempting to acquire process-lock: dotlock
 INFO environment: Acquired process lock: dotlock
 INFO environment: Released process lock: dotlock
 INFO environment: Acquired process lock: machine-action-5e8fabf089da6737431eeafdb74f5224
 INFO runner: Preparing hooks for middleware sequence...
 INFO runner: 3 hooks defined.
 INFO runner: Running action: machine_action_up #<Vagrant::Action::Builder:0x42fb910>
 INFO warden: Calling IN action: #<VagrantPlugins::ProviderVirtualBox::Action::CheckVirtualbox:0x43180f8>
DEBUG base: Windows. Trying VBOX_INSTALL_PATH for VBoxManage
DEBUG base: VBOX_INSTALL_PATH value: C:\Program Files\Oracle\VirtualBox\
 INFO base: VBoxManage path: C:\Program Files\Oracle\VirtualBox\VBoxManage.exe
 INFO subprocess: Starting process: ["C:\\Program Files\\Oracle\\VirtualBox\\VBoxManage.exe", "--version"]
 INFO subprocess: Command not in installer, restoring original environment...
DEBUG subprocess: Selecting on IO
DEBUG subprocess: stdout: 4.3.28r100309
DEBUG subprocess: Waiting for process to exit. Remaining to timeout: 32000
DEBUG subprocess: Exit status: 0
DEBUG meta: Finding driver for VirtualBox version: 4.3.28
 INFO meta: Using VirtualBox driver: VagrantPlugins::ProviderVirtualBox::Driver::Version_4_3
DEBUG base: Windows. Trying VBOX_INSTALL_PATH for VBoxManage
DEBUG base: VBOX_INSTALL_PATH value: C:\Program Files\Oracle\VirtualBox\
 INFO base: VBoxManage path: C:\Program Files\Oracle\VirtualBox\VBoxManage.exe
 INFO subprocess: Starting process: ["C:\\Program Files\\Oracle\\VirtualBox\\VBoxManage.exe", "list", "hostonlyifs"]
 INFO subprocess: Command not in installer, restoring original environment...
DEBUG subprocess: Selecting on IO
DEBUG subprocess: stdout: Name:            VirtualBox Host-Only Ethernet Adapter

GUID:            c31b6932-85ea-4377-9eb4-d1487f6f7fe2
DHCP:            Disabled
IPAddress:       192.168.56.1
NetworkMask:     255.255.255.0
IPV6Address:     fe80:0000:0000:0000:8c99:d131:2cf1:36bb
IPV6NetworkMaskPrefixLength: 128
HardwareAddress: 08:00:27:00:84:4e
MediumType:      Ethernet
Status:          Up
VBoxNetworkName: HostInterfaceNetworking-VirtualBox Host-Only Ethernet Adapter

DEBUG subprocess: Waiting for process to exit. Remaining to timeout: 31999
DEBUG subprocess: Exit status: 0
 INFO warden: Calling IN action: #<Vagrant::Action::Builtin::Call:0x43180e0>
 INFO runner: Preparing hooks for middleware sequence...
 INFO runner: 3 hooks defined.
 INFO runner: Running action: machine_action_up #<Vagrant::Action::Builder:0x2bf2510>
 INFO warden: Calling IN action: #<VagrantPlugins::ProviderVirtualBox::Action::Created:0x2e0fec0>
 INFO warden: Calling OUT action: #<VagrantPlugins::ProviderVirtualBox::Action::Created:0x2e0fec0>
 INFO runner: Preparing hooks for middleware sequence...
 INFO runner: 3 hooks defined.
 INFO runner: Running action: machine_action_up #<Vagrant::Action::Warden:0x2ef9548>
 INFO warden: Calling IN action: #<Proc:0x37ea2f0@c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:94 (lambda)>
 INFO warden: Calling IN action: #<Vagrant::Action::Builtin::HandleBox:0x2ef9458>
 INFO handle_box: Machine already has box. HandleBox will not run.
 INFO warden: Calling IN action: #<Proc:0x36857b8@c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:94 (lambda)>
 INFO warden: Calling IN action: #<VagrantPlugins::R10k::Modulegetter:0x4318098>

 INFO interface: info: vagrant-r10k: Building the r10k module path with puppet provisioner module_path "". (if module_path is an array, first element is used)
 INFO interface: info: ==> default: vagrant-r10k: Building the r10k module path with puppet provisioner module_path "". (if module_path is an array, first element is used)
==> default: vagrant-r10k: Building the r10k module path with puppet provisioner module_path "". (if module_path is an array, first element is used)
 INFO interface: info: vagrant-r10k: Building the r10k module path with puppet provisioner odule_path "". (if module_path is an array, first element is used)
 INFO interface: info: ==> default: vagrant-r10k: Building the r10k module path with puppet provisioner module_path "". (if module_path is an array, first element is used)
==> default: vagrant-r10k: Building the r10k module path with puppet provisioner module_path "". (if module_path is an array, first element is used)
ERROR warden: Error occurred: no implicit conversion of nil into String
 INFO warden: Beginning recovery process...
 INFO warden: Calling recover: #<Vagrant::Action::Builtin::Call:0x43180e0>
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO warden: Recovery complete.
ERROR warden: Error occurred: no implicit conversion of nil into String
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
ERROR warden: Error occurred: no implicit conversion of nil into String
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO environment: Released process lock: machine-action-5e8fabf089da6737431eeafdb74f5224
 INFO environment: Running hook: environment_unload
 INFO runner: Preparing hooks for middleware sequence...
 INFO runner: 3 hooks defined.
 INFO runner: Running action: environment_unload #<Vagrant::Action::Builder:0x454d430>
C:/Users/Jeff/.vagrant.d/gems/gems/vagrant-r10k-0.2.0/lib/vagrant-r10k/modulegetter.rb:90:in `join': no implicit conversion of nil into String (TypeError)
        from C:/Users/Jeff/.vagrant.d/gems/gems/vagrant-r10k-0.2.0/lib/vagrant-r10k/modulegetter.rb:90:in `call'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:95:in `block in finalize_action'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/builtin/handle_box.rb:56:in `call'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:95:in `block in finalize_action'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/builder.rb:116:in `call'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/runner.rb:66:in `block in run'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/util/busy.rb:19:in `busy'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/runner.rb:66:in `run'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/builtin/call.rb:53:in `call'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/plugins/providers/virtualbox/action/check_virtualbox.rb:17:in `call'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/builder.rb:116:in `call'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/runner.rb:66:in `block in run'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/util/busy.rb:19:in `busy'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/runner.rb:66:in `run'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/machine.rb:214:in `action_raw'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/machine.rb:191:in `block in action'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/environment.rb:516:in `lock'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/machine.rb:178:in `call'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/machine.rb:178:in `action'
        from c:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/batch_action.rb:82:in `block (2 levels) in run'
```

**Comment by jantman** [comment]
@jeff1evesque This appears to be because ``module_path`` is not configured for either the puppet provider or vagrant-r10k. This is really a bug because vagrant-r10k should detect this situation and issue an error.

Until then, if you set ``config.r10k.module_path`` to the path to your modules directory (see https://github.com/jantman/vagrant-r10k/blob/master/README.md#usage-with-explicit-path-to-module-installation-directory ) it should fix this issue.

**Comment by jeff1evesque** [comment]
@jantman, thank you for pointing me to your documentation.  I should have paid specific attention to your comment on the [usage](https://github.com/jantman/vagrant-r10k#usage) section:

>If you provide an array of directories in puppet.module_path, vagrant-r10k will use the first directory listed for auto configuration. If you want to let r10k use a different directory, see below.

Ironically, this is something I recognized on a previous project.  However, I'm not sure if you would like to qualify this issue as closed, as my problem has been resolved.
